### PR TITLE
Feature sorting columns with custom datetime formats

### PR DESCRIPTION
**Pull request by ferwe**
Added possibility to define a custom datetime format for sorting.

See https://datatables.net/blog/2014-12-18

### CODE REVIEW

File: Private/Script.Configuration.ps1 [hunk 1]
```diff
@@ -68,7 +68,7 @@ $Script:Configuration = [ordered] @{
             Header       = @{
                 CssLink = 'https://cdn.datatables.net/v/dt/jq-3.3.1/dt-1.10.18/af-2.3.2/b-1.5.4/b-colvis-1.5.4/b-html5-1.5.4/b-print-1.5.4/cr-1.5.0/fc-3.2.5/fh-3.1.4/kt-2.5.0/r-2.2.2/rg-1.1.0/rr-1.2.4/sc-1.5.0/sl-1.2.6/datatables.min.css'
                 Css     = "$PSScriptRoot\..\Resources\CSS\datatables.min.css"
-                JsLink  = "https://cdn.datatables.net/v/dt/jq-3.3.1/dt-1.10.18/af-2.3.2/b-1.5.4/b-colvis-1.5.4/b-html5-1.5.4/b-print-1.5.4/cr-1.5.0/fc-3.2.5/fh-3.1.4/kt-2.5.0/r-2.2.2/rg-1.1.0/rr-1.2.4/sc-1.5.0/sl-1.2.6/datatables.min.js"
+                JsLink  = "https://cdn.datatables.net/v/dt/jq-3.3.1/dt-1.10.18/af-2.3.2/b-1.5.4/b-colvis-1.5.4/b-html5-1.5.4/b-print-1.5.4/cr-1.5.0/fc-3.2.5/fh-3.1.4/kt-2.5.0/r-2.2.2/rg-1.1.0/rr-1.2.4/sc-1.5.0/sl-1.2.6/datatables.min.js","https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.8.4/moment.min.js","https://cdn.datatables.net/plug-ins/1.10.19/sorting/datetime-moment.js"
```
Comment (PrzemyslawKlys) [inline Private/Script.Configuration.ps1]: Do you need to have both? moment.min.js and datetime-moment.js? If yes, that's ok. Can you also download those files and put in "$PSScriptRoot\..\Resources\JS\datatables.min.js". JSLink is used by default, but people can choose to integrate code into HTML. Not advised but still an option.

File: Private/Script.Configuration.ps1 [hunk 2]
```diff
@@ -68,7 +68,7 @@ $Script:Configuration = [ordered] @{
...
                 JS      = "$PSScriptRoot\..\Resources\JS\datatables.min.js"
```
Comment (PrzemyslawKlys) [inline Private/Script.Configuration.ps1]: Please add files here that are local.

File: Private/Add-CustomFormatForDatetimeSorting.ps1 [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    #>
+    
+    [CmdletBinding()]
+    param(
+        [array]$DateTimeSortingFormat
+    )
+    if($Script:Output){Remove-Variable Output -Scope Script}
+
+    if($DateTimeSortingFormat){
+        [array]$Script:Output = foreach($format in $DateTimeSortingFormat){
```
Comment (PrzemyslawKlys) [inline Private/Add-CustomFormatForDatetimeSorting.ps1]: Changed this, You should use it this way += creates a new array on each ad.

File: Private/Add-CustomFormatForDatetimeSorting.ps1 [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            "$.fn.dataTable.moment( '$format' );"
```
Comment (PrzemyslawKlys) [inline Private/Add-CustomFormatForDatetimeSorting.ps1]: Don't you need to escape $ with `$ I believe I have seen issues when not escaped.

File: Private/Script.Configuration.ps1 (hunk 1 above)
Comment (ferwe) [reply Private/Script.Configuration.ps1]: yes both neede.
Ah yes, I didn't recognize the possibility for local ressources :)

Will fix it.

File: Private/Add-CustomFormatForDatetimeSorting.ps1 (hunk 4 above)
Comment (ferwe) [reply Private/Add-CustomFormatForDatetimeSorting.ps1]: I tested the behavior and it works great without escaping

File: Private/Script.Configuration.ps1 (hunk 2 above)
Comment (ferwe) [reply Private/Script.Configuration.ps1]: yep

**Comment by PrzemyslawKlys** [comment]
@ferwe can you provide some examples that users can use with New-HTMLTable and Out-HtmlView?

**Pull request merged**